### ISSUE TITLE
Respect YAML metadata blocks anywhere in the document

### ISSUE THREAD

**Issue by dpfu** [Zettlr/Zettlr#730]
## Description
Zettl renders a YAML metadata as ´YAML frontmatter´. As you can put YAML metadata anywhere in the document, Zettlr should be able to correctly respect and render those metadata blocks as well.

> A YAML metadata block may occur anywhere in the document, but if it is not at the beginning, it must be preceded by a blank line. [Pandoc Manual](https://pandoc.org/MANUAL.html#extension-yaml_metadata_block)

## Reproducing
Add a YAML metadata block in a document body (i.e. below a header).

## Expected behaviour
Zettlr should recognize YAML data anywhere in the document.

## Platform
 - OS and version: 10.14.05
 - Zettlr Version: 1.6.0.

## Additional information

My use-case for this: I want to be able to add TODOs as YAML metadata, as suggest [here](https://github.com/jgm/pandoc/issues/2552#issuecomment-159148164).

**Comment by nathanlesage** [comment]
See your point, but a case against this: These YAML metadata blocks are recognized by almost any parser at the beginning, but I'm not so sure about the in-document things. Secondly, I personally don't know if multiple YAML blocks align with minimalist and "correct" typesetting (something certainly debatable), and third, this would complicate the code further … if there are many people supporting this, I'll consider adding this, but only after some other issues have been resolved that are currently — attention, pun — bugging me …

**Comment by dpfu** [comment]
Thanks for the quick follow-up. Totally understand your point. It’s not super high priority, more the category “nice-to have”. 

(I just found your project last month, and I am already a fan and promoting Zettlr to friends, colleagues and students. Thanks and keep up
the good the work!)

**Comment by dpfu** [comment]
A quick workaround: If you have the same problem as me (Zettlr rendering YAML comments as large h1-headers), you an use a YAML block with YAML comments:

````
```
---
# My comment here
# more here
...
```

Those are completely ignored by Pandoc.

**Comment by nathanlesage** [comment]
Doesn't this also work?

```
```yaml
some yaml
\```
```

**Comment by dpfu** [comment]
Well, it works if you want a YAML block in your document. What I want is a comment block that is completely ignored by Pandoc (i.a. not written to the output doc). If you don't care for this you could also just use HTML-style `<!-- comments -->`.

**Comment by nathanlesage** [comment]
Well, the comments are indeed meant for such kinds of ToDos, why do you want to do this using a YAML block?

**Comment by dpfu** [comment]
Using a `YAML` block strips the comments when converting with Pandoc. This way,  they don't show up in the final HTML source (see [reference](https://github.com/jgm/pandoc/issues/2552#issuecomment-159148164)). Would you suggest a different approach? 

I have my markdown files in Git and I use [a workflow action](https://github.com/marketplace/actions/todo-to-issue) that generates (and closes) GitHub issues from my TODO comments. It works for me.

Thanks for your attention to this (rather minor) issue. I think we can close this, as it is maybe more a cosmetic issue at this point. Comments in YAML frontmatter are styled correctly, only in-document the show up as header. 

![Bildschirmfoto 2020-05-13 um 15 23 58](https://user-images.githubusercontent.com/1433348/81818186-000b5b80-952e-11ea-980a-4c6272c21843.png)
